hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieTableServiceClient.java

```diff
@@ -472,6 +472,18 @@ public HoodieWriteMetadata<O> cluster(String clusteringInstant, boolean shouldCo
     return clusteringMetadata;
   }
 
+  public boolean purgePendingClustering(String clusteringInstant) {
```

Comment:
do we need a separate method? Don't we already rollback pnding clustering before starting the clustering?
https://github.com/apache/hudi/blob/3f652e67637c31ef653689e7fa248a4ad072c535/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieTableServiceClient.java#L449

Reply:
incase of both compaction and clustering, rollback does not really delete the requested. 
Here the requirement is to completely rollback and delete the requested file as well. hence had to introduce a public method to write client to support this.